datalake/src/main/java/com/sequenceiq/datalake/flow/create/SdxCreateActions.java

```diff
@@ -260,4 +268,10 @@ protected Object getFailurePayload(SdxCreateFailedEvent payload, Optional<SdxCon
         };
     }
 
+    private void notifyEvent(SdxContext context, ResourceEvent resourceEvent) {
+        SdxCluster sdxCluster = sdxService.getById(context.getSdxId());
```

Comment:
It seems to me as though often the SdxCluster you grab here is already available when this method is called. As the getById is a database query I was wondering if it would make more sense to modify this method or have another method which accepts the cluster object itself, just to keep things clean and efficient.

The call may not be an efficiency issue at all though so if that is the case you can ignore this.

Or if for whatever reason the sdx cluster here is different than the one used in the other methods you can ignore this as well.

Reply:
That is not accurate. For the files where I think I have SdxCluster available, this method accepts SDXCluster as input to avoid database lookup. SdxDetachActions.java is one such example.

In the case of SdxCreateActions.java, the SdxCluster object is not available to pass.